# CCAA SPAIN DATA
Source: https://github.com/datadista/datasets/tree/master/COVID%2019

From May (~18-05-2020) **Datadista** was updating only the national dataset and not the data by autonomous community due to a change in the data collection methodology by the Ministry of Health. 
"The previous series is not comparable with the current one and the Carlos III Health Institute, the body in charge of publishing the historical series, has paralyzed the publication of data until they can correct it".

**Old historical series**: from 21-02-2020 to 18-05-2020

Update: 03-07-2020
The new historical series is now available in three data formats in the Datadista repository. They will keep the datasets from the previous historical series to allow comparison and analysis.

**New historical series**: from 31-01-2020 to 28-06-2020 (weekly update)

Source info **Instituto de Salud Carlos III**: https://cnecovid.isciii.es/covid19/#documentaci%C3%B3n-y-datos

The analysis of the epidemic begins when a cumulative incidence of 5 COVID-19 cases / 100,000 inhabitants is reached in each Autonomous Community and that initial day of analysis is assigned to all the provinces of the respective Autonomous Community.

In order to avoid inconsistencies in the Rt estimates, in the seven-day windows in which there is a majority of “0 cases”, the analysis is completed at the point of the last case + 7 days or at the end of the **epidemiological week of analysis** (whichever is earlier in time).



# R code to obtain the dataframe (old) 
Source Datadista: 
https://github.com/datadista/datasets/tree/master/COVID%2019


Cases: 
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_casos.csv

https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_casos_long.csv

Deaths:
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_fallecidos.csv

https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_fallecidos_long.csv


Hospitalized: https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_hospitalizados.csv

https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_hospitalizados_long.csv


Uci:
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_uci.csv

https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_uci_long.csv


Hospital discharge: 
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_altas.csv

https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_altas_long.csv

Autonomous community population: 
https://www.ine.es/jaxiT3/Tabla.htm?t=2853


Debido a un cambio en la metodología de recolección de datos del Ministerio de Sanidad, Datadista deja de actualizar los datos a nivel autonómico en el mes de mayo (dependiendo del tipo de dato:  los casos el 21-05-2020, los hospitalizados el 24-05-2020, los fallecidos el 24-05-2020, los de uci el 24-05-2020 y las altas el 18-05-2020). Esto es así por la imposibilidad de comparar las anteriores series históricas con las actuales debido al cambio. Solo actualizan los datos a nivel nacional y han paralizado la actualización autonómica hasta la corrección.

Como los datos menos actualizados son los de tipo altas, para la construcción de la tabla voy a seleccionar todos los tipos de datos hasta la fecha de actualización de altas el día 18-05-2020. 


In [ ]:
#Este código de R no funciona aquí, pero os lo dejo por si le quereis echar un vistazo:
#DATAFRAME HASTA EL 18-05-2020:
```{r}
library(tidyverse)
#CASOS
ccaa_casos_wider = read.csv("../datadista/ccaa_covid19_casos.csv") #actualizado hasta 21-05-2020
ccaa_casos_wider = ccaa_casos_wider[, 1:90] #selección hasta el 18-05-2020
casos_longer = pivot_longer(ccaa_casos_wider, cols = starts_with("X"), names_to = "fecha", values_to = "total")

casos_longer$fecha = gsub("X", "", casos_longer$fecha)
casos_longer$fecha = as.Date(gsub("[.]", "-", casos_longer$fecha))
casos_longer["tipo"] = "casos"


#HOSPITALIZADOS
ccaa_hospitalizados_wider = read.csv("../datadista/ccaa_covid19_hospitalizados.csv")
ccaa_hospitalizados_wider = ccaa_hospitalizados_wider[,1:90]
hospitalizados_longer = pivot_longer(ccaa_hospitalizados_wider, cols = starts_with("X"), names_to = "fecha", values_to = "total")

hospitalizados_longer$fecha = gsub("X", "", hospitalizados_longer$fecha)
hospitalizados_longer$fecha = as.Date(gsub("[.]", "-", hospitalizados_longer$fecha))
str(hospitalizados_longer)
hospitalizados_longer["tipo"] = "hospitalizados"


#FALLECIDOS
ccaa_fallecidos_wider = read.csv("../datadista/ccaa_covid19_fallecidos.csv")
ccaa_fallecidos_wider = ccaa_fallecidos_wider[,1:78]
dates = c("X2020.02.21", "X2020.02.22", "X2020.02.23", "X2020.02.24", "X2020.02.25", "X2020.02.26", "X2020.02.27", "X2020.02.28", "X2020.02.29","X2020.03.01", "X2020.03.02", "X2020.03.03")
ccaa_fallecidos_wider[dates] <- 0
ccaa_fallecidos_wider = ccaa_fallecidos_wider[, c(1:2, 79:90, 3:78)] #Ordenando las fechas

fallecidos_longer = pivot_longer(ccaa_fallecidos_wider, cols = starts_with("X"), names_to = "fecha", values_to = "total")

fallecidos_longer$fecha = gsub("X", "", fallecidos_longer$fecha)
fallecidos_longer$fecha = as.Date(gsub("[.]", "-", fallecidos_longer$fecha))
str(fallecidos_longer)
fallecidos_longer["tipo"] = "fallecidos"


#UCI
ccaa_uci_wider = read.csv("../datadista/ccaa_covid19_uci.csv")
ccaa_uci_wider = ccaa_uci_wider[,1:90]
uci_longer = pivot_longer(ccaa_uci_wider, cols = starts_with("X"), names_to = "fecha", values_to = "total")

uci_longer$fecha = gsub("X", "", uci_longer$fecha)
uci_longer$fecha = as.Date(gsub("[.]", "-", uci_longer$fecha))
str(uci_longer)
uci_longer["tipo"] = "uci"



#ALTAS
ccaa_altas_wider = read.csv("../datadista/ccaa_covid19_altas.csv")
ccaa_altas_wider = ccaa_altas_wider[, -3]
dates2 = c("X2020.02.21", "X2020.02.22", "X2020.02.23", "X2020.02.24", "X2020.02.25", "X2020.02.26", "X2020.02.27", "X2020.02.28", "X2020.02.29","X2020.03.01", "X2020.03.02", "X2020.03.03", "X2020.03.04", "X2020.03.05", "X2020.03.06", "X2020.03.07", "X2020.03.08")
ccaa_altas_wider[dates2] <- 0
ccaa_altas_wider = ccaa_altas_wider[, c(1:2, 74:90, 3:73)]


altas_longer = pivot_longer(ccaa_altas_wider, cols = starts_with("X"), names_to = "fecha", values_to = "total")

altas_longer$fecha = gsub("X", "", altas_longer$fecha)
altas_longer$fecha = as.Date(gsub("[.]", "-", altas_longer$fecha))
str(altas_longer)
altas_longer["tipo"] = "altas"


#UNIÓN DE TODAS LAS TABLAS
DatosCCAA = rbind(casos_longer, hospitalizados_longer, fallecidos_longer, uci_longer, altas_longer)
str(DatosCCAA)


#DATOS DE POBLACIÓN DE 2019 OBTENIDOS DEL INE (https://www.ine.es/jaxiT3/Datos.htm?t=2853#!tabs-tabla)
poblacion = read_tsv("../PoblacionCCAA.csv",locale = locale(grouping_mark = ".")) #para quitar los puntos que aparecen en el total de población que delimitan los miles
names(poblacion)[1] <- "Comunidades y Ciudades Autonomas"
poblacion = poblacion %>% 
  filter(`Comunidades y Ciudades Autonomas`!="Total",Sexo=="Total",Periodo==2019) %>%
  transmute(cod_ine=str_sub(`Comunidades y Ciudades Autonomas`,1,2),
    Poblacion_2019=Total,CCAA=`Comunidades y Ciudades Autonomas`)
str(poblacion)

poblacion$cod_ine = as.numeric(gsub("(?<![0-9])0+", "", poblacion$cod_ine, perl = TRUE)) #para quitar los ceros delante de los números del código ine


#UNIÓN POBLACIÓN Y DATOS CCAA
DatosCCAA_completo = poblacion %>% select(-CCAA) %>%
  inner_join(DatosCCAA,by="cod_ine")
DatosCCAA_completo = DatosCCAA_completo[, c(4, 1, 3, 5, 6, 2)]
str(DatosCCAA_completo)
#quitamos acentos y ñ:
DatosCCAA_completo$CCAA = as.factor(DatosCCAA_completo$CCAA)
levels(DatosCCAA_completo$CCAA) <- c("Andalucia", "Aragon", "Asturias", "Baleares",
                                     "C. Valenciana", "Canarias", "Cantabria", 
                                     "Castilla La Mancha", "Castilla y Leon", "Catalunya",
                                     "Ceuta", "Extremadura", "Galicia", "La Rioja", "Madrid",
                                     "Melilla", "Murcia", "Navarra", "Pais Vasco")
levels(DatosCCAA_completo$CCAA)


#GUARDAMOS EL DATASET
#Como objeto R
save(DatosCCAA_completo,
     file="../DatosCCAA_completo.Robj")

#Como archivo csv
write_csv(DatosCCAA_completo,
          path="../DatosCCAA_completo.csv")

#CAMBIO EL FORMATO A WIDER PARA QUE SE PUEDAN INCORPORAR MEJOR LOS DATOS DE TEMPERATURAS
datos = read.csv("../DatosCCAA_completo.csv")
datos_wider = pivot_wider(datos, names_from = "tipo", values_from = "total")
str(datos_wider)
#Guardamos como archivo csv
#Como archivo csv
write_csv(datos_wider,
          path="../DatosCCAA_completo_wider.csv")

```

In [ ]:
import pandas as pd
import codecs
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import datetime as dt
%matplotlib inline

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
datosCCAA = pd.read_csv('/content/DatosCCAA_completo_wider.csv')

In [ ]:
datosCCAA.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1672 entries, 0 to 1671
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   fecha           1672 non-null   object 
 1   cod_ine         1672 non-null   int64  
 2   CCAA            1672 non-null   object 
 3   Poblacion_2019  1672 non-null   int64  
 4   casos           1672 non-null   int64  
 5   hospitalizados  1344 non-null   float64
 6   fallecidos      1672 non-null   int64  
 7   uci             1374 non-null   float64
 8   altas           1667 non-null   float64
dtypes: float64(3), int64(4), object(2)
memory usage: 117.7+ KB


In [ ]:
datosCCAA.head(10)

fecha  cod_ine       CCAA  ...  fallecidos  uci  altas
0  2020-02-21        1  Andalucia  ...           0  NaN    0.0
1  2020-02-22        1  Andalucia  ...           0  NaN    0.0
2  2020-02-23        1  Andalucia  ...           0  NaN    0.0
3  2020-02-24        1  Andalucia  ...           0  NaN    0.0
4  2020-02-25        1  Andalucia  ...           0  NaN    0.0
5  2020-02-26        1  Andalucia  ...           0  NaN    0.0
6  2020-02-27        1  Andalucia  ...           0  NaN    0.0
7  2020-02-28        1  Andalucia  ...           0  NaN    0.0
8  2020-02-29        1  Andalucia  ...           0  NaN    0.0
9  2020-03-01        1  Andalucia  ...           0  NaN    0.0

[10 rows x 9 columns]

In [ ]:
datosCCAA.isna().sum()

fecha               0
cod_ine             0
CCAA                0
Poblacion_2019      0
casos               0
hospitalizados    328
fallecidos          0
uci               298
altas               5
dtype: int64

In [ ]:
datosCCAA[datosCCAA['hospitalizados'].isna()]

fecha  cod_ine       CCAA  ...  fallecidos  uci  altas
0     2020-02-21        1  Andalucia  ...           0  NaN    0.0
1     2020-02-22        1  Andalucia  ...           0  NaN    0.0
2     2020-02-23        1  Andalucia  ...           0  NaN    0.0
3     2020-02-24        1  Andalucia  ...           0  NaN    0.0
4     2020-02-25        1  Andalucia  ...           0  NaN    0.0
...          ...      ...        ...  ...         ...  ...    ...
1598  2020-03-06       19    Melilla  ...           0  NaN    0.0
1599  2020-03-07       19    Melilla  ...           0  NaN    0.0
1600  2020-03-08       19    Melilla  ...           0  NaN    0.0
1601  2020-03-09       19    Melilla  ...           0  0.0    NaN
1602  2020-03-10       19    Melilla  ...           0  0.0    NaN

[328 rows x 9 columns]

In [ ]:
datosCCAA.tail(10)

fecha  cod_ine     CCAA  ...  fallecidos  uci  altas
1662  2020-05-09       19  Melilla  ...           2  3.0  108.0
1663  2020-05-10       19  Melilla  ...           2  3.0  111.0
1664  2020-05-11       19  Melilla  ...           2  3.0  113.0
1665  2020-05-12       19  Melilla  ...           2  3.0  115.0
1666  2020-05-13       19  Melilla  ...           2  3.0  115.0
1667  2020-05-14       19  Melilla  ...           2  3.0  119.0
1668  2020-05-15       19  Melilla  ...           2  3.0  119.0
1669  2020-05-16       19  Melilla  ...           2  3.0  122.0
1670  2020-05-17       19  Melilla  ...           2  3.0  124.0
1671  2020-05-18       19  Melilla  ...           2  3.0  125.0

[10 rows x 9 columns]

In [ ]:
#Comprobación de que La Rioja sí tenía los datos del día 10 de mayo (luego se nos elimina)
datosCCAA[datosCCAA['fecha'] == '2020-05-10']

fecha  cod_ine                CCAA  ...  fallecidos     uci    altas
79    2020-05-10        1           Andalucia  ...        1320   745.0   9020.0
167   2020-05-10        2              Aragon  ...         824   255.0   3076.0
255   2020-05-10        3            Asturias  ...         283   121.0   1022.0
343   2020-05-10        4            Baleares  ...         209   165.0   1401.0
431   2020-05-10        5            Canarias  ...         148   178.0   1346.0
519   2020-05-10        6           Cantabria  ...         201    79.0   1933.0
607   2020-05-10        7     Castilla y Leon  ...        1897   546.0   7429.0
695   2020-05-10        8  Castilla La Mancha  ...        2759   623.0   6129.0
783   2020-05-10        9           Catalunya  ...        5532  2951.0  24349.0
871   2020-05-10       10       C. Valenciana  ...        1323   716.0   8861.0
959   2020-05-10       11         Extremadura  ...         480   109.0   2369.0
1047  2020-05-10       12             Galicia  ...         596   325.0   7347.0
1135  2020-05-10       13              Madrid  ...        8644  3520.0  39529.0
1223  2020-05-10       14              Murcia  ...         139   112.0   1782.0
1311  2020-05-10       15             Navarra  ...         488   135.0   3014.0
1399  2020-05-10       16          Pais Vasco  ...        1418   571.0  14646.0
1487  2020-05-10       17            La Rioja  ...         346    91.0   2648.0
1575  2020-05-10       18               Ceuta  ...           4     4.0    154.0
1663  2020-05-10       19             Melilla  ...           2     3.0    111.0

[19 rows x 9 columns]

In [ ]:
#datosCCAA['fecha'] = datosCCAA['fecha'].dt.strftime("%m/%d/%Y")

In [ ]:
#datosCCAA['fecha'] = pd.to_datetime(datosCCAA['fecha'], errors='coerce')

### Temperature csv

In [ ]:
temperatura = pd.read_csv('/content/nuevo_temperatura_completo.csv', index_col= 0)

In [ ]:
temperatura.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3507 entries, 0 to 3506
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   fecha      3507 non-null   object 
 1   comunidad  3507 non-null   object 
 2   tmed       3507 non-null   float64
 3   cod_ine    3507 non-null   int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 137.0+ KB


In [ ]:
temperatura.head()

fecha           comunidad  tmed  cod_ine
0  2020-01-01           ANDALUCIA  11.1        1
1  2020-01-01              ARAGON   2.8        2
2  2020-01-01            ASTURIAS   8.1        3
3  2020-01-01           CANTABRIA   8.0        6
4  2020-01-01  CASTILLA LA MANCHA   5.2        8

In [ ]:
temperatura.loc[(temperatura['fecha'] == '2020-02-22') & (temperatura['cod_ine']== 4)]

fecha       comunidad  tmed  cod_ine
1000  2020-02-22  ISLAS BALEARES  11.6        4

### Merge of CCAA data and temperature data

In [ ]:
datos = datosCCAA.merge(temperatura, on=["fecha", "cod_ine"])

In [ ]:
datos

fecha  cod_ine       CCAA  ...  altas  comunidad  tmed
0     2020-02-21        1  Andalucia  ...    0.0  ANDALUCIA  13.4
1     2020-02-22        1  Andalucia  ...    0.0  ANDALUCIA  14.1
2     2020-02-23        1  Andalucia  ...    0.0  ANDALUCIA  14.2
3     2020-02-24        1  Andalucia  ...    0.0  ANDALUCIA  14.0
4     2020-02-25        1  Andalucia  ...    0.0  ANDALUCIA  13.5
...          ...      ...        ...  ...    ...        ...   ...
1661  2020-05-14       19    Melilla  ...  119.0    MELILLA  18.4
1662  2020-05-15       19    Melilla  ...  119.0    MELILLA  18.0
1663  2020-05-16       19    Melilla  ...  122.0    MELILLA  19.0
1664  2020-05-17       19    Melilla  ...  124.0    MELILLA  17.7
1665  2020-05-18       19    Melilla  ...  125.0    MELILLA  19.2

[1666 rows x 11 columns]

El anterior dataset debería de tener 1672 filas (nº de filas del dataset original DatosCCAA_completo_wider.csv) pero tiene 1666. Voy a buscar las filas que faltan debido al merge con el dataset de temperatura y voy a añadirlas manualmente.

In [ ]:
del datos['comunidad']
datos

fecha  cod_ine       CCAA  ...  uci  altas  tmed
0     2020-02-21        1  Andalucia  ...  NaN    0.0  13.4
1     2020-02-22        1  Andalucia  ...  NaN    0.0  14.1
2     2020-02-23        1  Andalucia  ...  NaN    0.0  14.2
3     2020-02-24        1  Andalucia  ...  NaN    0.0  14.0
4     2020-02-25        1  Andalucia  ...  NaN    0.0  13.5
...          ...      ...        ...  ...  ...    ...   ...
1661  2020-05-14       19    Melilla  ...  3.0  119.0  18.4
1662  2020-05-15       19    Melilla  ...  3.0  119.0  18.0
1663  2020-05-16       19    Melilla  ...  3.0  122.0  19.0
1664  2020-05-17       19    Melilla  ...  3.0  124.0  17.7
1665  2020-05-18       19    Melilla  ...  3.0  125.0  19.2

[1666 rows x 10 columns]

In [ ]:
# datos['CCAA'] = datos['CCAA'].astype('category')
# datos['CCAA'].cat.categories

In [ ]:
datos.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1666 entries, 0 to 1665
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   fecha           1666 non-null   object 
 1   cod_ine         1666 non-null   int64  
 2   CCAA            1666 non-null   object 
 3   Poblacion_2019  1666 non-null   int64  
 4   casos           1666 non-null   int64  
 5   hospitalizados  1338 non-null   float64
 6   fallecidos      1666 non-null   int64  
 7   uci             1368 non-null   float64
 8   altas           1661 non-null   float64
 9   tmed            1666 non-null   float64
dtypes: float64(4), int64(4), object(2)
memory usage: 143.2+ KB


In [ ]:
datos.groupby('CCAA').count()

fecha  cod_ine  Poblacion_2019  ...  uci  altas  tmed
CCAA                                                ...                  
Andalucia              88       88              88  ...   71     88    88
Aragon                 88       88              88  ...   71     88    88
Asturias               88       88              88  ...   79     88    88
Baleares               88       88              88  ...   71     88    88
C. Valenciana          88       88              88  ...   88     88    88
Canarias               88       88              88  ...   71     88    88
Cantabria              88       88              88  ...   71     88    88
Castilla La Mancha     88       88              88  ...   71     88    88
Castilla y Leon        88       88              88  ...   71     88    88
Catalunya              88       88              88  ...   71     88    88
Ceuta                  88       88              88  ...   71     86    88
Extremadura            88       88              88  ...   71     88    88
Galicia                88       88              88  ...   71     88    88
La Rioja               87       87              87  ...   70     87    87
Madrid                 88       88              88  ...   71     87    88
Melilla                83       83              83  ...   66     81    83
Murcia                 88       88              88  ...   71     88    88
Navarra                88       88              88  ...   71     88    88
Pais Vasco             88       88              88  ...   71     88    88

[19 rows x 9 columns]

Como vemos, nos falta una fila de la comunidad de La Rioja (para llegar a las 88 en total) y 5 filas de la comunidad de Melilla (en las que no hay dato de temperatura y al hacer merge se nos han eliminado).


Vamos a identificar qué días son los que faltan y rellenarlos.

#### Rioja missing data

In [ ]:
Rioja = datos[datos['CCAA'] == 'La Rioja']
Rioja

fecha  cod_ine      CCAA  ...   uci   altas  tmed
1408  2020-02-21       17  La Rioja  ...   NaN     0.0  10.3
1409  2020-02-22       17  La Rioja  ...   NaN     0.0  12.4
1410  2020-02-23       17  La Rioja  ...   NaN     0.0  11.0
1411  2020-02-24       17  La Rioja  ...   NaN     0.0  12.0
1412  2020-02-25       17  La Rioja  ...   NaN     0.0  10.8
...          ...      ...       ...  ...   ...     ...   ...
1490  2020-05-14       17  La Rioja  ...  91.0  2867.0  13.6
1491  2020-05-15       17  La Rioja  ...  91.0  2927.0  13.0
1492  2020-05-16       17  La Rioja  ...  91.0  2990.0  13.8
1493  2020-05-17       17  La Rioja  ...  91.0  3048.0  15.4
1494  2020-05-18       17  La Rioja  ...  91.0  3107.0  17.0

[87 rows x 10 columns]

In [ ]:
Rioja['fecha'] = pd.to_datetime(Rioja['fecha'])

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
Rioja.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 87 entries, 1408 to 1494
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   fecha           87 non-null     datetime64[ns]
 1   cod_ine         87 non-null     int64         
 2   CCAA            87 non-null     object        
 3   Poblacion_2019  87 non-null     int64         
 4   casos           87 non-null     int64         
 5   hospitalizados  68 non-null     float64       
 6   fallecidos      87 non-null     int64         
 7   uci             70 non-null     float64       
 8   altas           87 non-null     float64       
 9   tmed            87 non-null     float64       
dtypes: datetime64[ns](1), float64(4), int64(4), object(1)
memory usage: 7.5+ KB


In [ ]:
#Faltan los datos del día 10 de mayo que sí que estaban presentes en el dataset original datosCCAA
pd.date_range(start = '2020-02-21', end = '2020-05-18' ).difference(Rioja.fecha)

DatetimeIndex(['2020-05-10'], dtype='datetime64[ns]', freq=None)

In [ ]:
Rioja.tail(10)

fecha  cod_ine      CCAA  ...   uci   altas  tmed
1485 2020-05-08       17  La Rioja  ...  91.0  2526.0  20.2
1486 2020-05-09       17  La Rioja  ...  91.0  2581.0  18.0
1487 2020-05-11       17  La Rioja  ...  91.0  2700.0  17.2
1488 2020-05-12       17  La Rioja  ...  91.0  2749.0  15.2
1489 2020-05-13       17  La Rioja  ...  91.0  2804.0  16.4
1490 2020-05-14       17  La Rioja  ...  91.0  2867.0  13.6
1491 2020-05-15       17  La Rioja  ...  91.0  2927.0  13.0
1492 2020-05-16       17  La Rioja  ...  91.0  2990.0  13.8
1493 2020-05-17       17  La Rioja  ...  91.0  3048.0  15.4
1494 2020-05-18       17  La Rioja  ...  91.0  3107.0  17.0

[10 rows x 10 columns]

In [ ]:
missing = {'fecha': ['2020-05-10'], 'cod_ine': [17], 'CCAA': ['La Rioja'], 'Poblacion_2019': [316798],
           'casos': [5378], 'hospitalizados': [1474.0], 'fallecidos': [346], 'uci': [91.0], 'altas': [1648.0],
           'tmed': [None]}

In [ ]:
rioja_missing = pd.DataFrame(data=missing)
rioja_missing

fecha  cod_ine      CCAA  ...   uci   altas  tmed
0  2020-05-10       17  La Rioja  ...  91.0  1648.0  None

[1 rows x 10 columns]

In [ ]:
rioja_missing.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   fecha           1 non-null      datetime64[ns]
 1   cod_ine         1 non-null      int64         
 2   CCAA            1 non-null      object        
 3   Poblacion_2019  1 non-null      int64         
 4   casos           1 non-null      int64         
 5   hospitalizados  1 non-null      float64       
 6   fallecidos      1 non-null      int64         
 7   uci             1 non-null      float64       
 8   altas           1 non-null      float64       
 9   tmed            0 non-null      float64       
dtypes: datetime64[ns](1), float64(4), int64(4), object(1)
memory usage: 208.0+ bytes


In [ ]:
rioja_missing['fecha'] = pd.to_datetime(rioja_missing['fecha'])

In [ ]:
rioja_missing['tmed'] = rioja_missing['tmed'].astype(float)

In [ ]:
rioja_missing

fecha  cod_ine      CCAA  Poblacion_2019  ...  fallecidos   uci   altas  tmed
0 2020-05-10       17  La Rioja          316798  ...         346  91.0  1648.0   NaN

[1 rows x 10 columns]

In [ ]:
datos['fecha'] = pd.to_datetime(datos['fecha'])

In [ ]:
datos.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1666 entries, 0 to 1665
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   fecha           1666 non-null   datetime64[ns]
 1   cod_ine         1666 non-null   int64         
 2   CCAA            1666 non-null   object        
 3   Poblacion_2019  1666 non-null   int64         
 4   casos           1666 non-null   int64         
 5   hospitalizados  1338 non-null   float64       
 6   fallecidos      1666 non-null   int64         
 7   uci             1368 non-null   float64       
 8   altas           1661 non-null   float64       
 9   tmed            1666 non-null   float64       
dtypes: datetime64[ns](1), float64(4), int64(4), object(1)
memory usage: 143.2+ KB


In [ ]:
#adding to datos the row is missing in La Rioja (day 2020-05-10)
datos_con_Rioja = datos.append(rioja_missing).reset_index()
datos_con_Rioja

index      fecha  cod_ine       CCAA  ...  fallecidos   uci   altas  tmed
0         0 2020-02-21        1  Andalucia  ...           0   NaN     0.0  13.4
1         1 2020-02-22        1  Andalucia  ...           0   NaN     0.0  14.1
2         2 2020-02-23        1  Andalucia  ...           0   NaN     0.0  14.2
3         3 2020-02-24        1  Andalucia  ...           0   NaN     0.0  14.0
4         4 2020-02-25        1  Andalucia  ...           0   NaN     0.0  13.5
...     ...        ...      ...        ...  ...         ...   ...     ...   ...
1662   1662 2020-05-15       19    Melilla  ...           2   3.0   119.0  18.0
1663   1663 2020-05-16       19    Melilla  ...           2   3.0   122.0  19.0
1664   1664 2020-05-17       19    Melilla  ...           2   3.0   124.0  17.7
1665   1665 2020-05-18       19    Melilla  ...           2   3.0   125.0  19.2
1666      0 2020-05-10       17   La Rioja  ...         346  91.0  1648.0   NaN

[1667 rows x 11 columns]

In [ ]:
datos_con_Rioja[datos_con_Rioja['CCAA'] == 'La Rioja']

index      fecha  cod_ine      CCAA  ...  fallecidos   uci   altas  tmed
1408   1408 2020-02-21       17  La Rioja  ...           0   NaN     0.0  10.3
1409   1409 2020-02-22       17  La Rioja  ...           0   NaN     0.0  12.4
1410   1410 2020-02-23       17  La Rioja  ...           0   NaN     0.0  11.0
1411   1411 2020-02-24       17  La Rioja  ...           0   NaN     0.0  12.0
1412   1412 2020-02-25       17  La Rioja  ...           0   NaN     0.0  10.8
...     ...        ...      ...       ...  ...         ...   ...     ...   ...
1491   1491 2020-05-15       17  La Rioja  ...         348  91.0  2927.0  13.0
1492   1492 2020-05-16       17  La Rioja  ...         348  91.0  2990.0  13.8
1493   1493 2020-05-17       17  La Rioja  ...         349  91.0  3048.0  15.4
1494   1494 2020-05-18       17  La Rioja  ...         351  91.0  3107.0  17.0
1666      0 2020-05-10       17  La Rioja  ...         346  91.0  1648.0   NaN

[88 rows x 11 columns]

#### Melilla missing data

In [ ]:
Melilla = datos[datos['CCAA'] == 'Melilla']
Melilla

fecha  cod_ine     CCAA  Poblacion_2019  ...  fallecidos  uci  altas  tmed
1583 2020-02-21       19  Melilla           86487  ...           0  NaN    0.0  16.0
1584 2020-02-22       19  Melilla           86487  ...           0  NaN    0.0  16.3
1585 2020-02-23       19  Melilla           86487  ...           0  NaN    0.0  15.8
1586 2020-02-24       19  Melilla           86487  ...           0  NaN    0.0  15.4
1587 2020-02-25       19  Melilla           86487  ...           0  NaN    0.0  16.6
...         ...      ...      ...             ...  ...         ...  ...    ...   ...
1661 2020-05-14       19  Melilla           86487  ...           2  3.0  119.0  18.4
1662 2020-05-15       19  Melilla           86487  ...           2  3.0  119.0  18.0
1663 2020-05-16       19  Melilla           86487  ...           2  3.0  122.0  19.0
1664 2020-05-17       19  Melilla           86487  ...           2  3.0  124.0  17.7
1665 2020-05-18       19  Melilla           86487  ...           2  3.0  125.0  19.2

[83 rows x 10 columns]

In [ ]:
#Faltan los datos de los días: 13, 15, 17, 18, 19 de abril que sí que estaban presentes en el dataset original datosCCAA
pd.date_range(start = '2020-02-21', end = '2020-05-18' ).difference(Melilla.fecha)

DatetimeIndex(['2020-04-13', '2020-04-15', '2020-04-17', '2020-04-18',
               '2020-04-19'],
              dtype='datetime64[ns]', freq=None)

In [ ]:
datosCCAA[(datosCCAA['CCAA'] == 'Melilla') & (datosCCAA['fecha'] == '2020-04-19')]

fecha  cod_ine     CCAA  ...  fallecidos  uci  altas
1642  2020-04-19       19  Melilla  ...           2  3.0   44.0

[1 rows x 9 columns]

In [ ]:
missing2 = {'fecha': ['2020-04-13','2020-04-15','2020-04-17', '2020-04-18', '2020-04-19'], 
            'cod_ine': [19, 19, 19, 19, 19], 
            'CCAA': ['Melilla','Melilla','Melilla','Melilla','Melilla'], 
            'Poblacion_2019': [86487, 86487, 86487, 86487, 86487],
            'casos': [101, 102, 104, 104, 107], 
            'hospitalizados': [43.0, 43.0, 44.0, 44.0, 44.0], 
            'fallecidos': [2,2,2,2,2], 
            'uci': [3.0, 3.0, 3.0, 3.0,3.0], 
            'altas': [18.0, 24.0, 30.0, 30.0,44.0],
            'tmed': [None, None, None, None, None]}

In [ ]:
melilla_missing = pd.DataFrame(data=missing2)
melilla_missing

fecha  cod_ine     CCAA  Poblacion_2019  ...  fallecidos  uci  altas  tmed
0  2020-04-13       19  Melilla           86487  ...           2  3.0   18.0  None
1  2020-04-15       19  Melilla           86487  ...           2  3.0   24.0  None
2  2020-04-17       19  Melilla           86487  ...           2  3.0   30.0  None
3  2020-04-18       19  Melilla           86487  ...           2  3.0   30.0  None
4  2020-04-19       19  Melilla           86487  ...           2  3.0   44.0  None

[5 rows x 10 columns]

In [ ]:
melilla_missing.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   fecha           5 non-null      datetime64[ns]
 1   cod_ine         5 non-null      int64         
 2   CCAA            5 non-null      object        
 3   Poblacion_2019  5 non-null      int64         
 4   casos           5 non-null      int64         
 5   hospitalizados  5 non-null      float64       
 6   fallecidos      5 non-null      int64         
 7   uci             5 non-null      float64       
 8   altas           5 non-null      float64       
 9   tmed            0 non-null      float64       
dtypes: datetime64[ns](1), float64(4), int64(4), object(1)
memory usage: 528.0+ bytes


In [ ]:
melilla_missing['fecha'] = pd.to_datetime(melilla_missing['fecha'])

In [ ]:
melilla_missing['tmed'] = melilla_missing['tmed'].astype(float)

In [ ]:
#adding to datos the row is missing in La Rioja (day 2020-05-10)
datos_complete = datos_con_Rioja.append(melilla_missing, ignore_index=True)

In [ ]:
datos_complete

index      fecha  cod_ine       CCAA  ...  fallecidos  uci  altas  tmed
0       0.0 2020-02-21        1  Andalucia  ...           0  NaN    0.0  13.4
1       1.0 2020-02-22        1  Andalucia  ...           0  NaN    0.0  14.1
2       2.0 2020-02-23        1  Andalucia  ...           0  NaN    0.0  14.2
3       3.0 2020-02-24        1  Andalucia  ...           0  NaN    0.0  14.0
4       4.0 2020-02-25        1  Andalucia  ...           0  NaN    0.0  13.5
...     ...        ...      ...        ...  ...         ...  ...    ...   ...
1667    NaN 2020-04-13       19    Melilla  ...           2  3.0   18.0   NaN
1668    NaN 2020-04-15       19    Melilla  ...           2  3.0   24.0   NaN
1669    NaN 2020-04-17       19    Melilla  ...           2  3.0   30.0   NaN
1670    NaN 2020-04-18       19    Melilla  ...           2  3.0   30.0   NaN
1671    NaN 2020-04-19       19    Melilla  ...           2  3.0   44.0   NaN

[1672 rows x 11 columns]

In [ ]:
datos_complete.tail(40)

index      fecha  cod_ine      CCAA  ...  fallecidos   uci   altas  tmed
1632  1632.0 2020-04-10       19   Melilla  ...           2   3.0    17.0  16.8
1633  1633.0 2020-04-11       19   Melilla  ...           2   3.0    18.0  15.6
1634  1634.0 2020-04-12       19   Melilla  ...           2   3.0    18.0  16.0
1635  1635.0 2020-04-14       19   Melilla  ...           2   3.0    18.0  16.4
1636  1636.0 2020-04-16       19   Melilla  ...           2   3.0    27.0  17.4
1637  1637.0 2020-04-20       19   Melilla  ...           2   3.0    46.0  17.1
1638  1638.0 2020-04-21       19   Melilla  ...           2   3.0    46.0  16.8
1639  1639.0 2020-04-22       19   Melilla  ...           2   3.0    50.0  18.4
1640  1640.0 2020-04-23       19   Melilla  ...           2   3.0    55.0  17.0
1641  1641.0 2020-04-24       19   Melilla  ...           2   3.0    61.0  16.7
1642  1642.0 2020-04-25       19   Melilla  ...           2   3.0    77.0  17.6
1643  1643.0 2020-04-26       19   Melilla  ...           2   3.0    81.0  18.2
1644  1644.0 2020-04-27       19   Melilla  ...           2   3.0    87.0  18.8
1645  1645.0 2020-04-28       19   Melilla  ...           2   3.0    92.0  18.5
1646  1646.0 2020-04-29       19   Melilla  ...           2   3.0    95.0  19.6
1647  1647.0 2020-04-30       19   Melilla  ...           2   3.0    96.0  19.5
1648  1648.0 2020-05-01       19   Melilla  ...           2   3.0    96.0  21.2
1649  1649.0 2020-05-02       19   Melilla  ...           2   3.0    96.0  22.6
1650  1650.0 2020-05-03       19   Melilla  ...           2   3.0    98.0  22.0
1651  1651.0 2020-05-04       19   Melilla  ...           2   3.0   101.0  21.8
1652  1652.0 2020-05-05       19   Melilla  ...           2   3.0   101.0  23.6
1653  1653.0 2020-05-06       19   Melilla  ...           2   3.0   104.0  23.3
1654  1654.0 2020-05-07       19   Melilla  ...           2   3.0   108.0  19.5
1655  1655.0 2020-05-08       19   Melilla  ...           2   3.0   108.0  20.0
1656  1656.0 2020-05-09       19   Melilla  ...           2   3.0   108.0  20.4
1657  1657.0 2020-05-10       19   Melilla  ...           2   3.0   111.0  19.2
1658  1658.0 2020-05-11       19   Melilla  ...           2   3.0   113.0  20.4
1659  1659.0 2020-05-12       19   Melilla  ...           2   3.0   115.0  18.0
1660  1660.0 2020-05-13       19   Melilla  ...           2   3.0   115.0  17.9
1661  1661.0 2020-05-14       19   Melilla  ...           2   3.0   119.0  18.4
1662  1662.0 2020-05-15       19   Melilla  ...           2   3.0   119.0  18.0
1663  1663.0 2020-05-16       19   Melilla  ...           2   3.0   122.0  19.0
1664  1664.0 2020-05-17       19   Melilla  ...           2   3.0   124.0  17.7
1665  1665.0 2020-05-18       19   Melilla  ...           2   3.0   125.0  19.2
1666     0.0 2020-05-10       17  La Rioja  ...         346  91.0  1648.0   NaN
1667     NaN 2020-04-13       19   Melilla  ...           2   3.0    18.0   NaN
1668     NaN 2020-04-15       19   Melilla  ...           2   3.0    24.0   NaN
1669     NaN 2020-04-17       19   Melilla  ...           2   3.0    30.0   NaN
1670     NaN 2020-04-18       19   Melilla  ...           2   3.0    30.0   NaN
1671     NaN 2020-04-19       19   Melilla  ...           2   3.0    44.0   NaN

[40 rows x 11 columns]

In [ ]:
datos_final = datos_complete.sort_values(by = ['fecha','cod_ine'], ignore_index=True)
datos_final

index      fecha  cod_ine        CCAA  ...  fallecidos    uci    altas  tmed
0        0.0 2020-02-21        1   Andalucia  ...           0    NaN      0.0  13.4
1       88.0 2020-02-21        2      Aragon  ...           0    NaN      0.0   8.8
2      176.0 2020-02-21        3    Asturias  ...           0    NaN      0.0  10.6
3      264.0 2020-02-21        4    Baleares  ...           0    NaN      0.0  11.1
4      352.0 2020-02-21        5    Canarias  ...           0    NaN      0.0  17.2
...      ...        ...      ...         ...  ...         ...    ...      ...   ...
1667  1319.0 2020-05-18       15     Navarra  ...         503  136.0   3905.0  15.0
1668  1407.0 2020-05-18       16  Pais Vasco  ...        1466  578.0  16160.0  14.0
1669  1494.0 2020-05-18       17    La Rioja  ...         351   91.0   3107.0  17.0
1670  1582.0 2020-05-18       18       Ceuta  ...           4    4.0    163.0  18.9
1671  1665.0 2020-05-18       19     Melilla  ...           2    3.0    125.0  19.2

[1672 rows x 11 columns]

In [ ]:
del datos_final['index']
datos_final

fecha  cod_ine        CCAA  ...    uci    altas  tmed
0    2020-02-21        1   Andalucia  ...    NaN      0.0  13.4
1    2020-02-21        2      Aragon  ...    NaN      0.0   8.8
2    2020-02-21        3    Asturias  ...    NaN      0.0  10.6
3    2020-02-21        4    Baleares  ...    NaN      0.0  11.1
4    2020-02-21        5    Canarias  ...    NaN      0.0  17.2
...         ...      ...         ...  ...    ...      ...   ...
1667 2020-05-18       15     Navarra  ...  136.0   3905.0  15.0
1668 2020-05-18       16  Pais Vasco  ...  578.0  16160.0  14.0
1669 2020-05-18       17    La Rioja  ...   91.0   3107.0  17.0
1670 2020-05-18       18       Ceuta  ...    4.0    163.0  18.9
1671 2020-05-18       19     Melilla  ...    3.0    125.0  19.2

[1672 rows x 10 columns]

Convert to complete csv file

In [ ]:
#datos_CCAA_hasta_mayo.csv
datos_final.to_csv('/content/datos_CCAA_hasta_mayo.csv',index=False, )

In [ ]:
prueba = pd.read_csv('/content/datos_fusion.csv')
prueba

index       fecha  cod_ine        CCAA  ...  fallecidos    uci    altas  tmed
0        0.0  2020-02-21        1   Andalucia  ...           0    NaN      0.0  13.4
1       88.0  2020-02-21        2      Aragon  ...           0    NaN      0.0   8.8
2      176.0  2020-02-21        3    Asturias  ...           0    NaN      0.0  10.6
3      264.0  2020-02-21        4    Baleares  ...           0    NaN      0.0  11.1
4      352.0  2020-02-21        5    Canarias  ...           0    NaN      0.0  17.2
...      ...         ...      ...         ...  ...         ...    ...      ...   ...
1667  1319.0  2020-05-18       15     Navarra  ...         503  136.0   3905.0  15.0
1668  1407.0  2020-05-18       16  Pais Vasco  ...        1466  578.0  16160.0  14.0
1669  1494.0  2020-05-18       17    La Rioja  ...         351   91.0   3107.0  17.0
1670  1582.0  2020-05-18       18       Ceuta  ...           4    4.0    163.0  18.9
1671  1665.0  2020-05-18       19     Melilla  ...           2    3.0    125.0  19.2

[1672 rows x 11 columns]

# R code to obtain the updated dataframe with temperatures (new)

Updated dataframes from Datadista:

Cases:
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_datos_isciii_nueva_serie.csv

Deaths:
https://github.com/datadista/datasets/blob/master/COVID%2019/ccaa_covid19_fallecidos_por_fecha_defuncion_nueva_serie.csv



In [ ]:
#Código en R:
```{r}
library(tidyverse)
#CASOS
ccaa_casos_longer = read.csv("../datadista2/ccaa_covid19_datos_isciii_nueva_serie.csv") #actualizado hasta 12-07-2020
str(ccaa_casos_longer)
ccaa_casos_longer$fecha = as.Date(ccaa_casos_longer$fecha)


#FALLECIDOS
ccaa_fallecidos_wider = read.csv("../datadista2/ccaa_covid19_fallecidos_por_fecha_defuncion_nueva_serie.csv")

dates = c("X2020.01.31", "X2020.02.01", "X2020.02.02", "X2020.02.03", "X2020.02.04", "X2020.02.05", "X2020.02.06", "X2020.02.07", "X2020.02.08","X2020.02.09", "X2020.02.10", "X2020.02.11","X2020.02.12")
ccaa_fallecidos_wider[dates] <- 0
ccaa_fallecidos_wider = ccaa_fallecidos_wider[, c(1:2, 158:170, 3:157)] #Ordenando las fechas: primero las dos primeras columnas (cod_ine y CCAA), después las introducidas al final, y por último el resto

fallecidos_longer = pivot_longer(ccaa_fallecidos_wider, cols = starts_with("X"), names_to = "fecha", values_to = "fallecidos")

fallecidos_longer$fecha = gsub("X", "", fallecidos_longer$fecha)
fallecidos_longer$fecha = as.Date(gsub("[.]", "-", fallecidos_longer$fecha))
str(fallecidos_longer)



#UNIÓN DE TODAS LAS TABLAS (DATASET ACTUALIZADO A 17-07-2020)
DatosCCAA_completo = inner_join(ccaa_casos_longer, fallecidos_longer,by=c("cod_ine", "fecha"))


#quitamos acentos y ñ:
DatosCCAA_completo$ccaa = as.factor(DatosCCAA_completo$ccaa)
levels(DatosCCAA_completo$ccaa)
levels(DatosCCAA_completo$ccaa) <- c("Andalucia", "Aragon", "Asturias", "Baleares",
                                     "C. Valenciana", "Canarias", "Cantabria", 
                                     "Castilla La Mancha", "Castilla y Leon", "Catalunya",
                                     "Ceuta", "Extremadura", "Galicia", "La Rioja", "Madrid",
                                     "Melilla", "Murcia", "Navarra", "Pais Vasco")


#GUARDAMOS EL DATASET
#Como objeto R
save(DatosCCAA_completo,
     file="../DatosCCAA_completo.Robj")

#Como archivo csv
write_csv(DatosCCAA_completo,
          path="../DatosCCAA_completo3.csv")
```

In [ ]:
datosCCAA_new = pd.read_csv('/content/DatosCCAA_completo3_sin_temphumedad.csv')
datosCCAA_new

fecha  cod_ine  ...           CCAA  fallecidos
0     2020-01-31        1  ...     Andalucía           0
1     2020-01-31        2  ...        Aragón           0
2     2020-01-31        3  ...       Asturias           0
3     2020-01-31        6  ...      Cantabria           0
4     2020-01-31       18  ...          Ceuta           0
...          ...      ...  ...            ...         ...
3111  2020-07-12       19  ...        Melilla           0
3112  2020-07-12       15  ...        Navarra           0
3113  2020-07-12       16  ...    País Vasco           1
3114  2020-07-12       17  ...       La Rioja           0
3115  2020-07-12       10  ...  C. Valenciana           0

[3116 rows x 10 columns]

In [ ]:
datosCCAA_new.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3116 entries, 0 to 3115
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   fecha                         3116 non-null   object
 1   cod_ine                       3116 non-null   int64 
 2   ccaa                          3116 non-null   object
 3   num_casos                     3116 non-null   int64 
 4   num_casos_prueba_pcr          3116 non-null   int64 
 5   num_casos_prueba_test_ac      3116 non-null   int64 
 6   num_casos_prueba_otras        3116 non-null   int64 
 7   num_casos_prueba_desconocida  3116 non-null   int64 
 8   CCAA                          3116 non-null   object
 9   fallecidos                    3116 non-null   int64 
dtypes: int64(7), object(3)
memory usage: 243.6+ KB


##### Temperature csv

In [ ]:
temperatura = pd.read_csv('/content/temperaturas_comunidad_ine.csv')
temperatura

time           community  temperature  cod_ine
0     2020-01-01           ANDALUCIA        13.46        1
1     2020-01-01              ARAGON        10.08        2
2     2020-01-01            ASTURIAS        10.29        3
3     2020-01-01           CANTABRIA        10.12        6
4     2020-01-01  CASTILLA LA MANCHA        12.54        8
...          ...                 ...          ...      ...
3662  2020-07-11      ISLAS CANARIAS        20.73        5
3663  2020-07-11            LA RIOJA        21.20       17
3664  2020-07-11             MELILLA        23.18       19
3665  2020-07-11              MURCIA        26.86       14
3666  2020-07-11          PAIS VASCO        18.48       16

[3667 rows x 4 columns]

##### Humidity csv

In [ ]:
humedad = pd.read_csv('/content/humedades_comunidad_ine.csv')
humedad

time           community  rel_humidity  cod_ine
0     2020-01-01           ANDALUCIA         71.14        1
1     2020-01-01              ARAGON         76.81        2
2     2020-01-01            ASTURIAS         75.98        3
3     2020-01-01           CANTABRIA         78.93        6
4     2020-01-01  CASTILLA LA MANCHA         80.63        8
...          ...                 ...           ...      ...
3662  2020-07-11      ISLAS CANARIAS         83.87        5
3663  2020-07-11            LA RIOJA         67.75       17
3664  2020-07-11             MELILLA         83.65       19
3665  2020-07-11              MURCIA         59.56       14
3666  2020-07-11          PAIS VASCO         76.90       16

[3667 rows x 4 columns]

#### Temperature and Humidity merge


In [ ]:
clima = temperatura.merge(humedad, on=["time", "cod_ine", "community"])
clima

time           community  temperature  cod_ine  rel_humidity
0     2020-01-01           ANDALUCIA        13.46        1         71.14
1     2020-01-01              ARAGON        10.08        2         76.81
2     2020-01-01            ASTURIAS        10.29        3         75.98
3     2020-01-01           CANTABRIA        10.12        6         78.93
4     2020-01-01  CASTILLA LA MANCHA        12.54        8         80.63
...          ...                 ...          ...      ...           ...
3662  2020-07-11      ISLAS CANARIAS        20.73        5         83.87
3663  2020-07-11            LA RIOJA        21.20       17         67.75
3664  2020-07-11             MELILLA        23.18       19         83.65
3665  2020-07-11              MURCIA        26.86       14         59.56
3666  2020-07-11          PAIS VASCO        18.48       16         76.90

[3667 rows x 5 columns]

In [ ]:
clima.head(50)

time             community  temperature  cod_ine  rel_humidity
0   2020-01-01             ANDALUCIA        13.46        1         71.14
1   2020-01-01                ARAGON        10.08        2         76.81
2   2020-01-01              ASTURIAS        10.29        3         75.98
3   2020-01-01             CANTABRIA        10.12        6         78.93
4   2020-01-01    CASTILLA LA MANCHA        12.54        8         80.63
5   2020-01-01       CASTILLA Y LEON        11.94        7         74.62
6   2020-01-01              CATALUÑA         7.79        9         79.76
7   2020-01-01                 CEUTA        11.35       18         85.80
8   2020-01-01   COMUNIDAD DE MADRID        11.11       13         80.30
9   2020-01-01  COMUNIDAD DE NAVARRA        10.09       15         87.51
10  2020-01-01  COMUNIDAD VALENCIANA        10.33       10         85.17
11  2020-01-01           EXTREMADURA        10.59       11         85.23
12  2020-01-01               GALICIA         9.81       12         84.88
13  2020-01-01        ISLAS BALEARES        10.90        4         79.65
14  2020-01-01        ISLAS CANARIAS        16.65        5         55.50
15  2020-01-01              LA RIOJA        12.51       17         66.85
16  2020-01-01               MELILLA        11.26       19         79.95
17  2020-01-01                MURCIA         9.49       14         85.88
18  2020-01-01            PAIS VASCO        10.06       16         78.47
19  2020-01-02             ANDALUCIA        13.20        1         74.81
20  2020-01-02                ARAGON         9.66        2         81.45
21  2020-01-02              ASTURIAS        11.19        3         73.43
22  2020-01-02             CANTABRIA        10.22        6         65.91
23  2020-01-02    CASTILLA LA MANCHA        12.11        8         79.31
24  2020-01-02       CASTILLA Y LEON        10.60        7         77.61
25  2020-01-02              CATALUÑA         7.80        9         82.75
26  2020-01-02                 CEUTA        11.28       18         89.45
27  2020-01-02   COMUNIDAD DE MADRID        10.02       13         76.38
28  2020-01-02  COMUNIDAD DE NAVARRA         8.13       15         90.06
29  2020-01-02  COMUNIDAD VALENCIANA         9.89       10         89.51
30  2020-01-02           EXTREMADURA         8.72       11         93.12
31  2020-01-02               GALICIA         9.86       12         83.77
32  2020-01-02        ISLAS BALEARES        11.33        4         83.70
33  2020-01-02        ISLAS CANARIAS        15.62        5         50.93
34  2020-01-02              LA RIOJA        11.02       17         75.01
35  2020-01-02               MELILLA        10.67       19         86.11
36  2020-01-02                MURCIA         8.90       14         90.86
37  2020-01-02            PAIS VASCO        10.35       16         68.76
38  2020-01-03             ANDALUCIA        12.86        1         76.38
39  2020-01-03                ARAGON         9.84        2         79.53
40  2020-01-03              ASTURIAS        10.02        3         78.22
41  2020-01-03             CANTABRIA         9.99        6         78.21
42  2020-01-03    CASTILLA LA MANCHA        10.18        8         85.93
43  2020-01-03       CASTILLA Y LEON         8.00        7         89.06
44  2020-01-03              CATALUÑA         7.85        9         80.89
45  2020-01-03                 CEUTA        12.03       18         82.47
46  2020-01-03   COMUNIDAD DE MADRID         8.70       13         86.82
47  2020-01-03  COMUNIDAD DE NAVARRA         8.47       15         92.75
48  2020-01-03  COMUNIDAD VALENCIANA        10.93       10         81.83
49  2020-01-03           EXTREMADURA         9.58       11         84.35

In [ ]:
clima = clima[['time', 'community', 'cod_ine', 'temperature', 'rel_humidity']]
clima

time           community  cod_ine  temperature  rel_humidity
0     2020-01-01           ANDALUCIA        1        13.46         71.14
1     2020-01-01              ARAGON        2        10.08         76.81
2     2020-01-01            ASTURIAS        3        10.29         75.98
3     2020-01-01           CANTABRIA        6        10.12         78.93
4     2020-01-01  CASTILLA LA MANCHA        8        12.54         80.63
...          ...                 ...      ...          ...           ...
3662  2020-07-11      ISLAS CANARIAS        5        20.73         83.87
3663  2020-07-11            LA RIOJA       17        21.20         67.75
3664  2020-07-11             MELILLA       19        23.18         83.65
3665  2020-07-11              MURCIA       14        26.86         59.56
3666  2020-07-11          PAIS VASCO       16        18.48         76.90

[3667 rows x 5 columns]

In [ ]:
clima.rename({'time' : 'fecha'}, axis=1, inplace=True)
clima.head()

fecha           community  cod_ine  temperature  rel_humidity
0  2020-01-01           ANDALUCIA        1        13.46         71.14
1  2020-01-01              ARAGON        2        10.08         76.81
2  2020-01-01            ASTURIAS        3        10.29         75.98
3  2020-01-01           CANTABRIA        6        10.12         78.93
4  2020-01-01  CASTILLA LA MANCHA        8        12.54         80.63

In [ ]:
clima.to_csv('/content/temperatura_humedad_juntos.csv')

### Merge of updated CCAA data and temperature  and humidity data

In [ ]:
datos_nuevos = datosCCAA_new.merge(clima, on=["fecha", "cod_ine"])
datos_nuevos

fecha  cod_ine  ... temperature  rel_humidity
0     2020-01-31        1  ...       15.04         77.30
1     2020-01-31        2  ...       15.18         78.11
2     2020-01-31        3  ...       16.58         76.16
3     2020-01-31        6  ...       15.61         75.73
4     2020-01-31       18  ...       15.83         57.52
...          ...      ...  ...         ...           ...
3092  2020-07-11       19  ...       23.18         83.65
3093  2020-07-11       15  ...       19.59         73.39
3094  2020-07-11       16  ...       18.48         76.90
3095  2020-07-11       17  ...       21.20         67.75
3096  2020-07-11       10  ...       25.23         67.83

[3097 rows x 13 columns]

In [ ]:
del datos_nuevos['community_x']
del datos_nuevos['CCAA']
datos_nuevos.head()

fecha  cod_ine       ccaa  ...  fallecidos  temperature  rel_humidity
0  2020-01-31        1  Andalucia  ...           0        15.04         77.30
1  2020-01-31        2     Aragon  ...           0        15.18         78.11
2  2020-01-31        3   Asturias  ...           0        16.58         76.16
3  2020-01-31        6  Cantabria  ...           0        15.61         75.73
4  2020-01-31       18      Ceuta  ...           0        15.83         57.52

[5 rows x 11 columns]

In [ ]:
datos_nuevos.tail()

fecha  cod_ine           ccaa  ...  fallecidos  temperature  rel_humidity
3092  2020-07-11       19        Melilla  ...           0        23.18         83.65
3093  2020-07-11       15        Navarra  ...           0        19.59         73.39
3094  2020-07-11       16     Pais Vasco  ...           0        18.48         76.90
3095  2020-07-11       17       La Rioja  ...           0        21.20         67.75
3096  2020-07-11       10  C. Valenciana  ...           0        25.23         67.83

[5 rows x 11 columns]

In [ ]:
datos_nuevos.groupby('ccaa').count()

fecha  cod_ine  ...  temperature  rel_humidity
ccaa                                ...                           
Andalucia             163      163  ...          163           163
Aragon                163      163  ...          163           163
Asturias              163      163  ...          163           163
Baleares              163      163  ...          163           163
C. Valenciana         163      163  ...          163           163
Canarias              163      163  ...          163           163
Cantabria             163      163  ...          163           163
Castilla La Mancha    163      163  ...          163           163
Castilla y Leon       163      163  ...          163           163
Catalunya             163      163  ...          163           163
Ceuta                 163      163  ...          163           163
Extremadura           163      163  ...          163           163
Galicia               163      163  ...          163           163
La Rioja              163      163  ...          163           163
Madrid                163      163  ...          163           163
Melilla               163      163  ...          163           163
Murcia                163      163  ...          163           163
Navarra               163      163  ...          163           163
Pais Vasco            163      163  ...          163           163

[19 rows x 10 columns]

In [ ]:
datos_nuevos.to_csv('/content/datos_temperatura_humedad.csv', index=False)

In [ ]:
prueba = pd.read_csv('/content/datos_temperatura_humedad.csv')
prueba

fecha  cod_ine           ccaa  ...  fallecidos  temperature  rel_humidity
0     2020-01-31        1      Andalucia  ...           0        15.04         77.30
1     2020-01-31        2         Aragon  ...           0        15.18         78.11
2     2020-01-31        3       Asturias  ...           0        16.58         76.16
3     2020-01-31        6      Cantabria  ...           0        15.61         75.73
4     2020-01-31       18          Ceuta  ...           0        15.83         57.52
...          ...      ...            ...  ...         ...          ...           ...
3092  2020-07-11       19        Melilla  ...           0        23.18         83.65
3093  2020-07-11       15        Navarra  ...           0        19.59         73.39
3094  2020-07-11       16     Pais Vasco  ...           0        18.48         76.90
3095  2020-07-11       17       La Rioja  ...           0        21.20         67.75
3096  2020-07-11       10  C. Valenciana  ...           0        25.23         67.83

[3097 rows x 11 columns]

**NO FALTA NINGUN DATO! 
LO QUE VIENE A CONTINUACIÓN ES LO QUE OCURRÍA CON LOS DATOS DE TEMPERATURA DE LA AEMET PERO CON LOS DATOS DE TEMPERATURA Y HUMEDAD DE ERA5 COPERNICUS NO FALTA NINGÚN DATO**

Faltan filas de las comunidades: **Ceuta**, La **Rioja** y **Melilla**. Todas las comunidades deben tener 157 filas cada una.

#### Rioja missing data


In [ ]:
temperatura.loc[(temperatura['fecha'] == '2020-05-10') & (temperatura['cod_ine']== 17)]

Empty DataFrame
Columns: [fecha, comunidad, tmed, cod_ine]
Index: []

In [ ]:
Rioja2 = datos_nuevos[datos_nuevos['ccaa'] == 'La Rioja']
Rioja2

fecha  cod_ine  ... fallecidos  tmed
17    2020-01-31       17  ...          0  13.0
36    2020-02-01       17  ...          0  13.6
55    2020-02-02       17  ...          0  13.8
74    2020-02-03       17  ...          0  13.9
93    2020-02-04       17  ...          0  11.2
...          ...      ...  ...        ...   ...
2897  2020-07-01       17  ...          0  23.4
2916  2020-07-02       17  ...          0  20.8
2935  2020-07-03       17  ...          0  18.1
2954  2020-07-04       17  ...          0  20.4
2973  2020-07-05       17  ...          0  24.4

[155 rows x 10 columns]

In [ ]:
Rioja2['fecha'] = pd.to_datetime(Rioja2['fecha'])

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
Rioja2.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 155 entries, 17 to 2973
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         155 non-null    datetime64[ns]
 1   cod_ine                       155 non-null    int64         
 2   ccaa                          155 non-null    object        
 3   num_casos                     155 non-null    int64         
 4   num_casos_prueba_pcr          155 non-null    int64         
 5   num_casos_prueba_test_ac      155 non-null    int64         
 6   num_casos_prueba_otras        155 non-null    int64         
 7   num_casos_prueba_desconocida  155 non-null    int64         
 8   fallecidos                    155 non-null    int64         
 9   tmed                          155 non-null    float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 13.3+ KB


In [ ]:
#Faltan los datos de los días 13 de febrero y 10 de mayo que sí que estaban presentes en el dataset original datosCCAA
pd.date_range(start = '2020-01-31', end = '2020-07-05' ).difference(Rioja2.fecha)

DatetimeIndex(['2020-02-13', '2020-05-10'], dtype='datetime64[ns]', freq=None)

In [ ]:
#Dichos días sí que están presentes en el dataset original. Es el dataset de temperatura el que no los tiene.
datosCCAA_new[(datosCCAA_new['fecha']=='2020-02-13') & (datosCCAA_new['cod_ine'] == 17)]

fecha  cod_ine  ...      CCAA  fallecidos
264  2020-02-13       17  ...  La Rioja           0

[1 rows x 10 columns]

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-05-10') & (datosCCAA_new['cod_ine'] == 17)]

fecha  cod_ine  ...      CCAA  fallecidos
1917  2020-05-10       17  ...  La Rioja           1

[1 rows x 10 columns]

In [ ]:
missing_rioja = {'fecha': ['2020-02-13', '2020-05-10'], 
                 'cod_ine': [17,17], 
                 'ccaa': ['La Rioja','La Rioja'], 
                 'num_casos': [0,2],
                 'num_casos_prueba_pcr': [0,2], 
                 'num_casos_prueba_test_ac': [0,0], 
                 'num_casos_prueba_otras': [0,0], 
                 'num_casos_prueba_desconocida': [0,0], 
                 'fallecidos': [0,1],
                 'tmed': [None, None]}

In [ ]:
missing_rioja = pd.DataFrame(data=missing_rioja)
missing_rioja

fecha  cod_ine  ... fallecidos  tmed
0  2020-02-13       17  ...          0  None
1  2020-05-10       17  ...          1  None

[2 rows x 10 columns]

In [ ]:
missing_rioja.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         2 non-null      datetime64[ns]
 1   cod_ine                       2 non-null      int64         
 2   ccaa                          2 non-null      object        
 3   num_casos                     2 non-null      int64         
 4   num_casos_prueba_pcr          2 non-null      int64         
 5   num_casos_prueba_test_ac      2 non-null      int64         
 6   num_casos_prueba_otras        2 non-null      int64         
 7   num_casos_prueba_desconocida  2 non-null      int64         
 8   fallecidos                    2 non-null      int64         
 9   tmed                          0 non-null      float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 288.0+ bytes


In [ ]:
missing_rioja['fecha'] = pd.to_datetime(missing_rioja['fecha'])

In [ ]:
missing_rioja['tmed'] = missing_rioja['tmed'].astype(float)

In [ ]:
datos_nuevos['fecha'] = pd.to_datetime(datos_nuevos['fecha'])

In [ ]:
datos_nuevos.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2975 entries, 0 to 2974
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         2975 non-null   datetime64[ns]
 1   cod_ine                       2975 non-null   int64         
 2   ccaa                          2975 non-null   object        
 3   num_casos                     2975 non-null   int64         
 4   num_casos_prueba_pcr          2975 non-null   int64         
 5   num_casos_prueba_test_ac      2975 non-null   int64         
 6   num_casos_prueba_otras        2975 non-null   int64         
 7   num_casos_prueba_desconocida  2975 non-null   int64         
 8   fallecidos                    2975 non-null   int64         
 9   tmed                          2975 non-null   float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 255.7+ KB


In [ ]:
#adding to datos the rows are missing in La Rioja (days 2020-02-13 and 2020-05-10)
datos_con_Rioja2 = datos_nuevos.append(missing_rioja, ignore_index=True)
datos_con_Rioja2

fecha  cod_ine  ... fallecidos  tmed
0    2020-01-31        1  ...          0  13.8
1    2020-01-31        2  ...          0  11.5
2    2020-01-31        3  ...          0  13.0
3    2020-01-31        6  ...          0  13.0
4    2020-01-31       18  ...          0  17.3
...         ...      ...  ...        ...   ...
2972 2020-07-05       16  ...          0  19.4
2973 2020-07-05       17  ...          0  24.4
2974 2020-07-05       10  ...          0  24.6
2975 2020-02-13       17  ...          0   NaN
2976 2020-05-10       17  ...          1   NaN

[2977 rows x 10 columns]

#### Ceuta missing data

In [ ]:
Ceuta = datos_nuevos[datos_nuevos['ccaa'] == 'Ceuta']
Ceuta

fecha  cod_ine   ccaa  ...  num_casos_prueba_desconocida  fallecidos  tmed
4    2020-01-31       18  Ceuta  ...                             0           0  17.3
23   2020-02-01       18  Ceuta  ...                             0           0  18.6
42   2020-02-02       18  Ceuta  ...                             0           0  15.6
61   2020-02-03       18  Ceuta  ...                             0           0  16.4
80   2020-02-04       18  Ceuta  ...                             0           0  16.6
...         ...      ...    ...  ...                           ...         ...   ...
2884 2020-07-01       18  Ceuta  ...                             0           0  26.8
2903 2020-07-02       18  Ceuta  ...                             0           0  26.8
2922 2020-07-03       18  Ceuta  ...                             0           0  27.0
2941 2020-07-04       18  Ceuta  ...                             0           0  22.6
2960 2020-07-05       18  Ceuta  ...                             0           0  24.0

[156 rows x 10 columns]

In [ ]:
Ceuta['fecha'] = pd.to_datetime(Ceuta['fecha'])

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
Ceuta.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 156 entries, 4 to 2960
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         156 non-null    datetime64[ns]
 1   cod_ine                       156 non-null    int64         
 2   ccaa                          156 non-null    object        
 3   num_casos                     156 non-null    int64         
 4   num_casos_prueba_pcr          156 non-null    int64         
 5   num_casos_prueba_test_ac      156 non-null    int64         
 6   num_casos_prueba_otras        156 non-null    int64         
 7   num_casos_prueba_desconocida  156 non-null    int64         
 8   fallecidos                    156 non-null    int64         
 9   tmed                          156 non-null    float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 13.4+ KB


In [ ]:
#Faltan los datos del día 30 de junio que sí que estaban presentes en el dataset original datosCCAA
pd.date_range(start = '2020-01-31', end = '2020-07-05' ).difference(Ceuta.fecha)

DatetimeIndex(['2020-06-30'], dtype='datetime64[ns]', freq=None)

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-06-30') & (datosCCAA_new['cod_ine'] == 18)]

fecha  cod_ine  ...   CCAA  fallecidos
2873  2020-06-30       18  ...  Ceuta           0

[1 rows x 10 columns]

In [ ]:
missing_ceuta = {'fecha': ['2020-06-30'], 
                 'cod_ine': [18], 
                 'ccaa': ['Ceuta'], 
                 'num_casos': [0],
                 'num_casos_prueba_pcr': [0], 
                 'num_casos_prueba_test_ac': [0], 
                 'num_casos_prueba_otras': [0], 
                 'num_casos_prueba_desconocida': [0], 
                 'fallecidos': [0],
                 'tmed': [None]}

In [ ]:
missing_ceuta = pd.DataFrame(data=missing_ceuta)
missing_ceuta

fecha  cod_ine   ccaa  ...  num_casos_prueba_desconocida  fallecidos  tmed
0  2020-06-30       18  Ceuta  ...                             0           0  None

[1 rows x 10 columns]

In [ ]:
missing_ceuta.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         1 non-null      datetime64[ns]
 1   cod_ine                       1 non-null      int64         
 2   ccaa                          1 non-null      object        
 3   num_casos                     1 non-null      int64         
 4   num_casos_prueba_pcr          1 non-null      int64         
 5   num_casos_prueba_test_ac      1 non-null      int64         
 6   num_casos_prueba_otras        1 non-null      int64         
 7   num_casos_prueba_desconocida  1 non-null      int64         
 8   fallecidos                    1 non-null      int64         
 9   tmed                          0 non-null      float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 208.0+ bytes


In [ ]:
missing_ceuta['fecha'] = pd.to_datetime(missing_ceuta['fecha'])

In [ ]:
missing_ceuta['tmed'] = missing_ceuta['tmed'].astype(float)

In [ ]:
#adding to datos_con_Rioja2 the rows are missing in Ceuta (day 2020-06-30)
datos_con_ceuta = datos_con_Rioja2.append(missing_ceuta, ignore_index=True)
datos_con_ceuta

fecha  cod_ine  ... fallecidos  tmed
0    2020-01-31        1  ...          0  13.8
1    2020-01-31        2  ...          0  11.5
2    2020-01-31        3  ...          0  13.0
3    2020-01-31        6  ...          0  13.0
4    2020-01-31       18  ...          0  17.3
...         ...      ...  ...        ...   ...
2973 2020-07-05       17  ...          0  24.4
2974 2020-07-05       10  ...          0  24.6
2975 2020-02-13       17  ...          0   NaN
2976 2020-05-10       17  ...          1   NaN
2977 2020-06-30       18  ...          0   NaN

[2978 rows x 10 columns]

#### Melilla missing data

In [ ]:
Melilla2 = datos_nuevos[datos_nuevos['ccaa'] == 'Melilla']
Melilla2

fecha  cod_ine  ... fallecidos  tmed
14   2020-01-31       19  ...          0  19.0
33   2020-02-01       19  ...          0  17.4
52   2020-02-02       19  ...          0  16.5
71   2020-02-03       19  ...          0  15.0
90   2020-02-04       19  ...          0  13.1
...         ...      ...  ...        ...   ...
2894 2020-07-01       19  ...          0  29.2
2913 2020-07-02       19  ...          0  31.0
2932 2020-07-03       19  ...          0  27.8
2951 2020-07-04       19  ...          0  24.6
2970 2020-07-05       19  ...          0  25.0

[152 rows x 10 columns]

In [ ]:
Melilla2['fecha'] = pd.to_datetime(Melilla2['fecha'])

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [ ]:
Melilla2.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 152 entries, 14 to 2970
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         152 non-null    datetime64[ns]
 1   cod_ine                       152 non-null    int64         
 2   ccaa                          152 non-null    object        
 3   num_casos                     152 non-null    int64         
 4   num_casos_prueba_pcr          152 non-null    int64         
 5   num_casos_prueba_test_ac      152 non-null    int64         
 6   num_casos_prueba_otras        152 non-null    int64         
 7   num_casos_prueba_desconocida  152 non-null    int64         
 8   fallecidos                    152 non-null    int64         
 9   tmed                          152 non-null    float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 13.1+ KB


In [ ]:
#Faltan los datos de los días 13,15,17,18,19 de abril que sí que estaban presentes en el dataset original datosCCAA
pd.date_range(start = '2020-01-31', end = '2020-07-05' ).difference(Melilla2.fecha)

DatetimeIndex(['2020-04-13', '2020-04-15', '2020-04-17', '2020-04-18',
               '2020-04-19'],
              dtype='datetime64[ns]', freq=None)

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-04-13') & (datosCCAA_new['cod_ine'] == 19)]

fecha  cod_ine  ...     CCAA  fallecidos
1401 2020-04-13       19  ...  Melilla           0

[1 rows x 10 columns]

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-04-15') & (datosCCAA_new['cod_ine'] == 19)]

fecha  cod_ine  ...     CCAA  fallecidos
1439 2020-04-15       19  ...  Melilla           0

[1 rows x 10 columns]

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-04-17') & (datosCCAA_new['cod_ine'] == 19)]

fecha  cod_ine  ...     CCAA  fallecidos
1477 2020-04-17       19  ...  Melilla           0

[1 rows x 10 columns]

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-04-18') & (datosCCAA_new['cod_ine'] == 19)]

fecha  cod_ine  ...     CCAA  fallecidos
1496 2020-04-18       19  ...  Melilla           0

[1 rows x 10 columns]

In [ ]:
datosCCAA_new[(datosCCAA_new['fecha']=='2020-04-19') & (datosCCAA_new['cod_ine'] == 19)]

fecha  cod_ine  ...     CCAA  fallecidos
1515 2020-04-19       19  ...  Melilla           0

[1 rows x 10 columns]

In [ ]:
missing_melilla = {'fecha': ['2020-04-13','2020-04-15','2020-04-17', '2020-04-18', '2020-04-19'], 
            'cod_ine': [19, 19, 19, 19, 19], 
            'ccaa': ['Melilla','Melilla','Melilla','Melilla','Melilla'], 
            'num_casos': [1, 0, 0, 0, 0],
            'num_casos_prueba_pcr': [1, 0, 0, 0, 0], 
            'num_casos_prueba_test_ac': [0, 0, 0, 0, 0], 
            'num_casos_prueba_otras': [0,0,0,0,0], 
            'num_casos_prueba_desconocida': [0, 0, 0, 0,0], 
            'fallecidos': [0, 0, 0, 0,0],
            'tmed': [None, None, None, None, None]}

In [ ]:
missing_melilla2 = pd.DataFrame(data=missing_melilla)
missing_melilla2

fecha  cod_ine     ccaa  ...  num_casos_prueba_desconocida  fallecidos  tmed
0  2020-04-13       19  Melilla  ...                             0           0  None
1  2020-04-15       19  Melilla  ...                             0           0  None
2  2020-04-17       19  Melilla  ...                             0           0  None
3  2020-04-18       19  Melilla  ...                             0           0  None
4  2020-04-19       19  Melilla  ...                             0           0  None

[5 rows x 10 columns]

In [ ]:
missing_melilla2.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 10 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   fecha                         5 non-null      datetime64[ns]
 1   cod_ine                       5 non-null      int64         
 2   ccaa                          5 non-null      object        
 3   num_casos                     5 non-null      int64         
 4   num_casos_prueba_pcr          5 non-null      int64         
 5   num_casos_prueba_test_ac      5 non-null      int64         
 6   num_casos_prueba_otras        5 non-null      int64         
 7   num_casos_prueba_desconocida  5 non-null      int64         
 8   fallecidos                    5 non-null      int64         
 9   tmed                          0 non-null      float64       
dtypes: datetime64[ns](1), float64(1), int64(7), object(1)
memory usage: 528.0+ bytes


In [ ]:
missing_melilla2['fecha'] = pd.to_datetime(missing_melilla2['fecha'])

In [ ]:
missing_melilla2['tmed'] = missing_melilla2['tmed'].astype(float)

In [ ]:
#adding to datos_con_ceuta the rows are missing in Melila
datos_con_melilla = datos_con_ceuta.append(missing_melilla2, ignore_index=True)
datos_con_melilla

fecha  cod_ine  ... fallecidos  tmed
0    2020-01-31        1  ...          0  13.8
1    2020-01-31        2  ...          0  11.5
2    2020-01-31        3  ...          0  13.0
3    2020-01-31        6  ...          0  13.0
4    2020-01-31       18  ...          0  17.3
...         ...      ...  ...        ...   ...
2978 2020-04-13       19  ...          0   NaN
2979 2020-04-15       19  ...          0   NaN
2980 2020-04-17       19  ...          0   NaN
2981 2020-04-18       19  ...          0   NaN
2982 2020-04-19       19  ...          0   NaN

[2983 rows x 10 columns]

In [ ]:
datos_total = datos_con_melilla.sort_values(by = ['fecha','cod_ine'], ignore_index=True)
datos_total

fecha  cod_ine  ... fallecidos  tmed
0    2020-01-31        1  ...          0  13.8
1    2020-01-31        2  ...          0  11.5
2    2020-01-31        3  ...          0  13.0
3    2020-01-31        4  ...          0  13.8
4    2020-01-31        5  ...          0  19.9
...         ...      ...  ...        ...   ...
2978 2020-07-05       15  ...          0  21.8
2979 2020-07-05       16  ...          0  19.4
2980 2020-07-05       17  ...          0  24.4
2981 2020-07-05       18  ...          0  24.0
2982 2020-07-05       19  ...          0  25.0

[2983 rows x 10 columns]

Convert to complete csv file

In [ ]:
datos_total.to_csv('/content/datos_CCAA_actualizados.csv',index=False)

In [ ]:
prueba3 = pd.read_csv('/content/datos_CCAA_actualizados.csv')
prueba3

fecha  cod_ine  ... fallecidos  tmed
0     2020-01-31        1  ...          0  13.8
1     2020-01-31        2  ...          0  11.5
2     2020-01-31        3  ...          0  13.0
3     2020-01-31        4  ...          0  13.8
4     2020-01-31        5  ...          0  19.9
...          ...      ...  ...        ...   ...
2978  2020-07-05       15  ...          0  21.8
2979  2020-07-05       16  ...          0  19.4
2980  2020-07-05       17  ...          0  24.4
2981  2020-07-05       18  ...          0  24.0
2982  2020-07-05       19  ...          0  25.0

[2983 rows x 10 columns]